builtin/providers/openstack/resource_openstack_lb_pool_v1.go

```diff
@@ -278,6 +279,16 @@ func resourceLBPoolV1Delete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error deleting OpenStack LB Pool: %s", err)
 	}
 
+	for {
+		_, err := pools.Get(networkingClient, d.Id()).Extract()
+		if err == nil {
+			log.Printf("[DEBUG] Pool (%s) is not deleted yet, waiting for 1 second", d.Id())
+			time.Sleep(1000 * time.Millisecond)
+		} else {
+			break;
+		}
+	}
+
```

Comment:
We generally wrap operations like this in a timeout to prevent infinite loops. Luckily there's a helper function you can use:

https://github.com/hashicorp/terraform/blob/master/helper/resource/wait.go#L12

Here's an example usage:

https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_autoscaling_group.go#L461-L476